### PR TITLE
WIP feature #6297 - wallet on desktop

### LINKED ISSUE #6297: [EPIC] Port Wallet to Desktop
### User Story

As a user I want Wallet on Desktop, because... Wallet on Desktop.

### Description

As the very first version, we'll just try to port the mobile Wallet directly as it is to desktop and only solve the most severe responsiveness issues. This is merely an experiment to see if it would work at all.

- [ ] port 2 panel designs to desktop
- [ ] Implement chat commands on desktop #6945
- [ ] transaction history (http calls don't work on desktop, at least not on linux) - status-im/react-native-desktop#373
- [ ] set recipient by scanning qr (camera on desktop not supported)
- [ ] offline detection - always says it's offline on linux, so it's being ignored by wallet when on desktop

base

### PR DESCRIPTION
**Pull request by goranjovic**
fixes #6297

### Summary:

Wallet on Desktop. Added a tab, which displays main wallet screen in main area. Sidebar is empty.
Basic wallet functionality is also there, with some noted exceptions.

What works:
- signing phrase onboarding
- display of balance and tokens in main wallet screen
- adding/removing enabled tokens
- send transaction to contact or by address
- show own address qr in receive transaction
- set custom gas settings

What does not work:
- send a transaction request (we have no chat commands on desktop yet)
- transaction history (http calls don't work on desktop, at least not on linux) - status-im/react-native-desktop#373
- set recipient by scanning qr (camera on desktop not supported)
- generally this PR does only main Wallet tab, not chat commands and not dapp transactions.
- network switching, only mainnet
- offline detection - always says it's offline on linux, so it's being ignored by wallet when on desktop

Notable responsiveness issues:
- missing title in Assets selection screen (the one where you enable/disable tokens)
- blue area below token list in main wallet screen
- missing component that displays menu popups - using dialog as a short term solution
- checkbox component is weird on desktop

## Testing notes (optional):

All platforms should be tested - shared code was affected.

#### Platforms (optional)
- Android
- iOS
- macOS
- Linux

#### Areas that maybe impacted (optional)
**Functional**
- wallet / transactions

status: ready

### CODE REVIEW

**Comment by goranjovic** [comment]
screenshot from 2018-10-15 11-36-20


screenshot from 2018-10-15 12-21-02

screenshot from 2018-10-15 12-21-16

screenshot from 2018-10-15 13-04-48

screenshot from 2018-10-15 12-46-14

screenshot from 2018-10-15 12-46-31

**Comment by chadyj** [comment]
Woah great one Goran!

Take a look @EugeOrtiz @denis-sharypin @errorists @andmironov 

Should we redesign wallet to work with 2 panes, or perhaps look at a desktop layout that supports 1 pane? Would be great to 1) get this looking decent as a first step, and 2) look at best design solutions for the longer term.

**Comment by goranjovic** [comment]
Also keep in mind that we'll be implementing Wallet Redesign soon enough, so it makes little sense to micro-optimize the existing layout for desktop.  On the other hand, we do want the new screens to look good when big.

**Comment by chadyj** [comment]
> Also keep in mind that we'll be implementing Wallet Redesign soon enough, so it makes little sense to micro-optimize the existing layout for desktop

In that case what about making the left panel blue so the contrast isn't so jarring?

**Comment by goranjovic** [comment]
Good idea, but to be fair, what to put in the sidebar is a good question in old and new designs alike.

**Review by dmitryn — APPROVED** [review]

**Comment by yenda** [comment]
Did you try getting rid of the conditional in `init.core`?
```
            #(when-not platform/desktop?
               (initialize-wallet %))
```

**Comment by goranjovic** [comment]
@yenda Haven't even noticed it :D Thanks!

**Comment by chadyj** [comment]
@goranjovic I added you to the Desktop sync tomorrow so we can discuss Wallet and getting this merged. Also @corpetty is joining to discuss desktop security.

**Comment by chadyj** [comment]
We touched on wallet in the desktop sync, but didn't get a chance to hear from @MaxRis @siphiuel @vkjr about whether we should include Wallet in Desktop, most likely behind a "dev mode" flag.  

The reason Wallet is important for Desktop is that it allows our primary use case of sending assets to contacts, but more importantly, it will allow SNT use cases like moderation, tribute to talk, and more. Status without SNT isn't really Status, so I'd advocate that we include it for visibility, and transparency.  Like everything else in Desktop it is a work in progress, but having it will focus attention in much the same way Desktop has done for chat.

**Comment by maxhora** [comment]
I agree that Wallet is essential part. Especially, if to compare with competitors. Can I help somehow to speedup things for merging this in without breaking others functionality?

**Comment by vitvly** [comment]
Agree with a proposal to include it under a dev flag in Desktop.

**Comment by goranjovic** [comment]
Created an issue to add dev mode and network switching to desktop #6477

**Comment by vkjr** [comment]
Looks like I was thinking about this but forget to say in voice on the meeting :)
I think that work on wallet will distract focus point from polishing chat, since it takes not only dev, but also QA and design resources. But if we have enough of them, I think adding it under DEV is a good solution.

File: src/status_im/ui/components/checkbox/styles.cljs [hunk 1]
```diff
@@ -18,9 +19,11 @@
           :width         26
           :height        26}))
 
-(def check-icon
-  {:width  12
-   :height 12})
+(defstyle check-icon
+  {:width   12
+   :height  12
+   ;;TODO(goranjovic) - this is a quickfix because checkboxes are invisible on desktop
```
Comment (vkjr) [inline src/status_im/ui/components/checkbox/styles.cljs:25]: @goranjovic, is it a bug in `react-native-desktop`? Would you mind adding corresponding issue to rnd repo?

File: src/status_im/ui/components/toolbar/styles.cljs [hunk 2]
```diff
@@ -85,9 +87,20 @@
 
 (def item-text-white-background {:color colors/blue})
 
-;;TODO(goranjovic) - Breaks the toolbar title into new line on smaller screens
-;;e.g. see Discover > Popular hashtags on iPhone 5s
-(def ios-content-item {:position :absolute :right 40 :left 40})
+;;TODO(goranjovic) - This doesn't work at all on desktop. Review why we even need absolute position hacks
```
Comment (vkjr) [inline src/status_im/ui/components/toolbar/styles.cljs:90]: Creating `react-native-desktop` issue with a small js example representing a problem would be very helpful.

File: src/status_im/ui/screens/wallet/main/styles.cljs [hunk 3]
```diff
@@ -120,7 +122,8 @@
    :color     colors/black})
 
 (def asset-item-currency
-  {:font-size   16
+  {:flex 1
```
Comment (vkjr) [inline src/status_im/ui/screens/wallet/main/styles.cljs:132]: Incorrect indentation

File: src/status_im/ui/screens/wallet/send/views.cljs [hunk 4]
```diff
@@ -141,7 +142,8 @@
   (let [{:keys [amount amount-text amount-error asset-error show-password-input? to to-name sufficient-funds?
                 sufficient-gas? in-progress? from-chat? symbol]} transaction
         {:keys [decimals] :as token} (tokens/asset-for (ethereum/network->chain-keyword network) symbol)
-        online? (= :online network-status)]
+        ;; TODO(goranjovic): offline status detection doesn't work on desktop yet
```
Comment (vkjr) [inline src/status_im/ui/screens/wallet/send/views.cljs:148]: @siphiuel, weren't we fixing this?

**Review by vkjr — APPROVED** [review]

File: src/status_im/ui/components/toolbar/styles.cljs (hunk 2 above)
Comment (goranjovic) [reply src/status_im/ui/components/toolbar/styles.cljs:90]: @vkjr This one is entirely `status-react` issue. At some point we used an absolute positioning hack to fix centering on some iphones. This is that issue coming back at us with vengeance.

File: src/status_im/ui/components/checkbox/styles.cljs (hunk 1 above)
Comment (goranjovic) [reply src/status_im/ui/components/checkbox/styles.cljs:25]: @vkjr yep, adding them all

**Comment by denis-sharypin** [comment]
Guys, @goranjovic , @chadyj  I would prefer to have such layout for desktop's wallet. Wdyt?

base

**Comment by goranjovic** [comment]
@denis-sharypin Just wondering if this idea is a reaction to the given fact that desktop ui already has a sidebar or if there are another reason to go with it?

@siphiuel I think I remember you or someone else mentioning that the reason sidebar was there in the first place is some issue with modals?

**Comment by chadyj** [comment]
That UI looks great and seems to be a way we can bridge mobile and desktop through the 2 pane approach (and is conceptually similar to chat) with the least amount of effort with great results.

What would the default screen be on the right pane?

**Comment by vitvly** [comment]
@goranjovic Yes, but this issue has been recently resolved: https://github.com/status-im/react-native-desktop/issues/326

**Comment by goranjovic** [comment]
@chadyj What I'm trying to figure out is if we even want to stick with the "two pane" solution, given that the original reason for it no longer exists.

**Comment by chadyj** [comment]
Pro of 2 pane is it's compatible with mobile implementation and doesn't require a dedicated redesign.

Con? No designs. Double implementation?

**Comment by goranjovic** [comment]
Con is parallel implementation of a part of navigation. Because there are two levels of navigation on desktop, the outer one for the left pane and the inner one for the right one.

On mobile there is only one screen area, and thus only one navigation.

If we go with two panes, we will need separate navigation logic on both platforms.

**Comment by denis-sharypin** [comment]
But this job was already done for the desktop chat?

**Comment by goranjovic** [comment]
The only thing I want to avoid is us implementing the two panels in Wallet only to double back and remove two panels altogether. That's why I'm raising the question.

**Comment by chadyj** [comment]
@denis-sharypin do you have plans to redesign wallet to 2 panels?
Or are you committed to this approach? https://github.com/status-im/status-react/pull/6312#issuecomment-442043419

Big decisions!

**Comment by denis-sharypin** [comment]
As was discussed on our wallet sync call we are implementing two panels design here https://github.com/status-im/status-react/pull/6312#issuecomment-442043419

**Comment by chadyj** [comment]
Goran shall we close this and revisit once the redesigns are in mobile?
We can create a new EPIC and scope out todo's.

**Pull request closed without merging**